### PR TITLE
Country - fetch data

### PR DESCRIPTION
**Pull request by shreeyash07**
- Addresses 
- #37 
- #86 

## Changes
- Fetched country details data
- Fetched disaggregation data
- small css changes

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] codegen errors
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

### CODE REVIEW

File: app/views/Dashboard/Country/index.tsx [hunk 1]
```diff
@@ -102,13 +134,20 @@ interface Props {
...
+
 function Country(props: Props) {
     const {
         filterValues,
     } = props;
 
-    const variables = useMemo(() => ({
-        iso3: filterValues?.country ?? 'AFG',
+    const countryVariables = useMemo(() => ({
+        iso3: filterValues?.country ?? 'NPL',
```
Comment (subinasr) [inline app/views/Dashboard/Country/index.tsx]: Why? I thought they wanted a country on top of the list to be selected as default, which is 'AFG'.
If this is a temporary thing, please add a 'NOTE' to remind you to revert it back in the final stage.

**Review by priyesh777 — APPROVED** [review]

File: app/views/Dashboard/Country/index.tsx [hunk 2]
```diff
@@ -180,7 +284,60 @@ function Country(props: Props) {
...
+
+    const genders = useMemo(() => (
+        unique(
+            indicatorResponse?.disaggregation.genderDisaggregation ?? [],
+            (d) => d.category,
+        ).map((item, index) => (
+            {
+                key: `${item.category}${item.indicatorValue}`,
+                gender: item.category,
+                fill: COLORS[index % COLORS.length] ?? 'green',
```
Comment (subinasr) [inline app/views/Dashboard/Country/index.tsx]: Why 'green'? Please use relevant colors.

File: app/views/Dashboard/Country/index.tsx [hunk 3]
```diff
@@ -180,7 +284,60 @@ function Country(props: Props) {
...
+
+    const age = useMemo(() => (
+        unique(
+            indicatorResponse?.disaggregation.ageDisaggregation ?? [],
+            (d) => d.category,
+        ).map((item, index) => (
+            {
+                key: `${item.category}${item.indicatorValue}`,
+                age: item.category,
+                fill: COLORS[index % COLORS.length] ?? 'green',
```
Comment (subinasr) [inline app/views/Dashboard/Country/index.tsx]: same here

File: app/views/Dashboard/Country/index.tsx [hunk 4]
```diff
@@ -102,13 +134,20 @@ interface Props {
     filterValues?: FilterType | undefined;
 }
 
+interface legend {
```
Comment (subinasr) [inline app/views/Dashboard/Country/index.tsx]:
```suggestion
interface Legend {
```

File: app/views/Dashboard/Country/index.tsx [hunk 5]
```diff
@@ -180,7 +284,60 @@ function Country(props: Props) {
...
+                            className={styles.legendBox}
+                            style={{ backgroundColor: item.fill }}
+                        />
+                        <div className={styles.legendDetails}>
+                            {item.age
+                                && <div>{item.age}</div>}
+                            {item.gender
+                                && <div>{item.gender}</div>}
+                            <div className={styles.regionalText}>
+                                Regional 30%
```
Comment (subinasr) [inline app/views/Dashboard/Country/index.tsx]: Let's add a note here so that we don't forget to put in regional data from the API.

**Review by subinasr — CHANGES_REQUESTED** [review]

File: app/views/Dashboard/Country/index.tsx [hunk 6]
```diff
@@ -144,32 +212,68 @@ function Country(props: Props) {
...
+        decimalToPercentage(countryResponse?.countryProfile.internetAccessRegion)
+    ), [countryResponse?.countryProfile.internetAccessRegion]);
+
+    const regional = useMemo(() => (
+        decimalToPercentage(countryResponse?.countryProfile.newCasesRegionShare)
+    ), [countryResponse?.countryProfile.newCasesRegionShare]);
+
+    const countryWiseOutbreakCases: countryWiseOutbreakCases[] | undefined = useMemo(() => {
+        const casesGroupList = listToGroupList(
+            contextResponse?.contextualData || [],
```
Comment (AdityaKhatri) [inline app/views/Dashboard/Country/index.tsx]:
```suggestion
            contextResponse?.contextualData ?? [],
```

File: app/views/Dashboard/Overview/OverviewTable/index.tsx [hunk 7]
```diff
@@ -66,7 +66,7 @@ function countryListCell(props: CountryListCellProps) {
     );
 }
 function MyCell<T>(props: TableCellProps<T>) {
-    const { className, ...otherProps } = props;
+    const { ...otherProps } = props;
```
Comment (AdityaKhatri) [inline app/views/Dashboard/Overview/OverviewTable/index.tsx]: Lets, tara lets not.

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: app/views/Dashboard/CombinedIndicators/styles.css [hunk 8]
```diff
@@ -4,17 +4,23 @@
     justify-content: center;
     gap: var(--dui-spacing-super-large);
 
+    .combined-indicator-header {
+        background-color: white;
+        padding: var(--spacing);
```
Comment (priyesh777) [inline app/views/Dashboard/CombinedIndicators/styles.css:9]: I guess the padding here doesn't make sense

**Review by priyesh777 — APPROVED** [review]

File: app/views/Dashboard/Country/index.tsx [hunk 9]
```diff
@@ -144,43 +176,111 @@ function Country(props: Props) {
...
+        const casesGroupList = listToGroupList(
+            countryResponse?.contextualData ?? [],
+            (emergency) => emergency.emergency,
+        );
+        const findLatestDate = (items: EmergencyItems[]) => {
+            const date = items.map((element) => element.contextDate);
+            const latestDate = date.reduce((acc, item) => (item > acc ? item : acc));
+            const indexOfLatestDate = date.indexOf(latestDate);
+            return items[indexOfLatestDate];
+        };
```
Comment (AdityaKhatri) [inline app/views/Dashboard/Country/index.tsx]: sort((a, b) => compareDate(a.contextDate, b.contextDate))

File: app/views/Dashboard/Overview/OverviewTable/index.tsx [hunk 10]
```diff
@@ -65,14 +65,16 @@ function countryListCell(props: CountryListCellProps) {
...
     );
 }
+
 function MyCell<T>(props: TableCellProps<T>) {
-    const { className, ...otherProps } = props;
+    const { className, value } = props;
     return (
-        <div className={_cs(otherProps.value && +(otherProps.value) > 40 && styles.highIndicator)}>
-            <TableCell {...otherProps} />
+        <div className={_cs(className, (value && +(value) > 40 && styles.highIndicator))}>
```
Comment (AdityaKhatri) [inline app/views/Dashboard/Overview/OverviewTable/index.tsx]: put logic outside

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: app/views/Dashboard/Country/index.tsx [hunk 11]
```diff
@@ -144,43 +168,112 @@ function Country(props: Props) {
...
+    ), [countryResponse?.countryProfile.newCasesRegionShare]);
+
+    const countryWiseOutbreakCases: CountryWiseOutbreakCases[] | undefined = useMemo(() => {
+        const casesGroupList = listToGroupList(
+            countryResponse?.contextualData ?? [],
+            (emergency) => emergency.emergency,
+        );
+
+        const latestDateItems = (items: EmergencyItems[]) => {
+            items.sort((a, b) => Number(a.contextDate) - Number(b.contextDate));
```
Comment (AdityaKhatri) [inline app/views/Dashboard/Country/index.tsx]:
```suggestion
            items.sort((a, b) => compareDate(a.contextDate, b.contextDate));
```

File: app/components/CustomLegend/index.tsx [hunk 12]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { _cs } from '@togglecorp/fujs';
+import styles from './styles.css';
+
+export interface Props {
+    className?: string;
+    age?: string
+    genders?: string;
```
Comment (tnagorra) [inline app/components/CustomLegend/index.tsx]:
```suggestion
    gender?: string;
```

File: app/components/CustomLegend/index.tsx [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+            <div
+                className={styles.legendBox}
+                style={{ backgroundColor: fill }}
+            />
+            <div className={styles.legendDetails}>
+                {age
+                    && <div>{age}</div>}
+                {genders
+                    && <div>{genders}</div>}
+                {/* FIX ME: Add regional data */}
```
Comment (tnagorra) [inline app/components/CustomLegend/index.tsx]:
```suggestion
                {/* FIXME: Add regional data */}
```

File: app/views/Dashboard/CombinedIndicators/styles.css [hunk 14]
```diff
@@ -4,17 +4,23 @@
     justify-content: center;
     gap: var(--dui-spacing-super-large);
 
+    .combined-indicator-header {
+        background-color: white;
```
Comment (tnagorra) [inline app/views/Dashboard/CombinedIndicators/styles.css]: Let's use a css variable

File: app/views/Dashboard/Country/index.tsx [hunk 15]
```diff
@@ -48,24 +46,40 @@ interface ScoreCardProps {
     metricType: 'positive' | 'negative';
     indicator?: 'red' | 'yellow' | 'orange' | 'green' | undefined;
 }
-interface countryWiseOutbreakCases {
+interface CountryWiseOutbreakCases extends EmergencyItems {
     key: string;
+}
+interface EmergencyItems {
```
Comment (tnagorra) [inline app/views/Dashboard/Country/index.tsx:53]: Lets move this before CountryWiseOutbreakCases

File: app/views/Dashboard/Country/index.tsx [hunk 16]
```diff
@@ -81,36 +95,47 @@ const COUNTRY_PROFILE = gql`
...
-    filterValues?: FilterType | undefined;
-}
-
 function Country(props: Props) {
     const {
         filterValues,
     } = props;
 
-    const variables = useMemo(() => ({
+    const countryVariables = useMemo(() => ({
```
Comment (tnagorra) [inline app/views/Dashboard/Country/index.tsx]:
```suggestion
    const countryVariables = useMemo((): CountryQueryVariables => ({
```

File: app/views/Dashboard/Country/index.tsx [hunk 17]
```diff
@@ -144,43 +169,112 @@ function Country(props: Props) {
...
+        decimalToPercentage(countryResponse?.countryProfile.newCasesRegionShare)
+    ), [countryResponse?.countryProfile.newCasesRegionShare]);
+
+    const countryWiseOutbreakCases: CountryWiseOutbreakCases[] | undefined = useMemo(() => {
+        const casesGroupList = listToGroupList(
+            countryResponse?.contextualData ?? [],
+            (emergency) => emergency.emergency,
+        );
+
+        const latestDateItems = (items: EmergencyItems[]) => {
```
Comment (tnagorra) [inline app/views/Dashboard/Country/index.tsx]:
```suggestion
        const getLatestDateItems = (items: EmergencyItems[]) => {
```

File: app/views/Dashboard/Country/index.tsx [hunk 18]
```diff
@@ -144,43 +169,112 @@ function Country(props: Props) {
...
+            items.sort((a, b) => compareDate(a.contextDate, b.contextDate, -1));
```
Comment (tnagorra) [inline app/views/Dashboard/Country/index.tsx]: `sort` mutates the list so let's make a copy

File: app/views/Dashboard/Country/index.tsx [hunk 19]
```diff
@@ -144,43 +169,112 @@ function Country(props: Props) {
...
+
+            return items[0];
+        };
+
+        return (
+            Object.entries(casesGroupList).map(([emergency, emergencyItems]) => (
+                {
+                    ...latestDateItems(emergencyItems),
+                    key: `${emergency}`,
```
Comment (tnagorra) [inline app/views/Dashboard/Country/index.tsx]:
```suggestion
                    key: emergency,
```

File: app/views/Dashboard/Country/index.tsx [hunk 20]
```diff
@@ -144,43 +169,112 @@ function Country(props: Props) {
...
         );
         return mapToList(outbreakGroupList,
             (group, key) => group.reduce(
                 (acc, item) => ({
                     ...acc,
                     [item.emergency]: item.contextIndicatorValue,
+                    date: new Date(item.contextDate)
+                        .toLocaleString(
+                            'default', { month: 'short' },
+                        ),
```
Comment (tnagorra) [inline app/views/Dashboard/Country/index.tsx]: Let's define a utility function

File: app/views/Dashboard/Country/index.tsx [hunk 21]
```diff
@@ -144,43 +169,112 @@ function Country(props: Props) {
...
                 }), { date: key },
             ));
-    }, [countryResponse?.countryEmergencyProfile]);
+    }, [countryResponse?.contextualData]);
 
     const outbreaks = useMemo(() => (
-        unique(countryResponse?.countryEmergencyProfile ?? [],
-            (d) => d.emergency).map((item) => {
+        unique(
```
Comment (tnagorra) [inline app/views/Dashboard/Country/index.tsx:238]: Let's define a FIXME to remove these unique

File: app/views/Dashboard/Country/index.tsx [hunk 22]
```diff
@@ -144,43 +169,112 @@ function Country(props: Props) {
...
                 emergency: item.emergency,
-                fill: colors[item.emergency] ?? 'pink',
+                fill: colors[item.emergency] ?? '#FFDD98',
             });
         })
-    ), [countryResponse?.countryEmergencyProfile]);
+    ), [countryResponse?.contextualData]);
+
+    const genders: CustomLegendProps[] = useMemo(() => (
+        unique(
```
Comment (tnagorra) [inline app/views/Dashboard/Country/index.tsx:256]: Let's define a FIXME to remove these unique

File: app/views/Dashboard/Country/index.tsx [hunk 23]
```diff
@@ -144,43 +169,112 @@ function Country(props: Props) {
...
+            countryResponse?.disaggregation.genderDisaggregation ?? [],
+            (d) => d.category,
+        ).map((item, index) => (
+            {
+                key: `${item.category}${item.indicatorValue}`,
```
Comment (tnagorra) [inline app/views/Dashboard/Country/index.tsx]:
```suggestion
                key: `${item.category}-${item.indicatorValue}`,
```

File: app/views/Dashboard/Country/index.tsx [hunk 24]
```diff
@@ -144,43 +169,112 @@ function Country(props: Props) {
...
+        ))
+    ), [countryResponse?.disaggregation.genderDisaggregation]);
+
+    const age: CustomLegendProps[] = useMemo(() => (
+        unique(
+            countryResponse?.disaggregation.ageDisaggregation ?? [],
+            (d) => d.category,
+        ).map((item, index) => (
+            {
+                key: `${item.category}${item.indicatorValue}`,
```
Comment (tnagorra) [inline app/views/Dashboard/Country/index.tsx]:
```suggestion
                key: `${item.category}-${item.indicatorValue}`,
```

File: app/views/Dashboard/Country/index.tsx [hunk 25]
```diff
@@ -248,6 +342,24 @@ function Country(props: Props) {
         indicator: metricTypeForColor(data),
     }), [metricTypeForColor]);
 
+    const customLegendParams = useCallback((_, data: CustomLegendProps) => ({
+        age: data.age,
+        genders: data.gender,
+        fill: data.fill,
+    }), []);
+
+    const ageCustomLegend = (legendData: CustomLegendProps[]) => (
```
Comment (tnagorra) [inline app/views/Dashboard/Country/index.tsx]: We do not need to create a function.

File: app/views/Dashboard/Country/index.tsx [hunk 26]
```diff
@@ -437,9 +550,12 @@ function Country(props: Props) {
...
                                     <>
                                         {`${internetAccess}%`}
-                                        <div className={styles.regionalText}>
-                                            Regional 30%
-                                        </div>
+                                        <TextOutput
+                                            labelContainerClassName={styles.regionalText}
+                                            valueContainerClassName={styles.regionalText}
+                                            label={countryResponse?.countryProfile.region}
+                                            value={`${internetAccessRegion}%`}
```
Comment (tnagorra) [inline app/views/Dashboard/Country/index.tsx]: Let's check if this value is defined or not.

File: app/views/Dashboard/Country/index.tsx [hunk 27]
```diff
@@ -484,13 +606,19 @@ function Country(props: Props) {
...
-                                            Regional 29%
-                                        </div>
+
+                                        <TextOutput
+                                            labelContainerClassName={styles.regionalText}
+                                            valueContainerClassName={styles.regionalText}
+                                            label={countryResponse?.countryProfile.region}
+                                            value={(countryResponse
+                                                ?.countryProfile.medicalStaffRegion)
+                                                ?.toFixed(0)}
```
Comment (tnagorra) [inline app/views/Dashboard/Country/index.tsx]:
```suggestion
                                                ?.toFixed(2)}
```

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: app/components/CustomLegend/index.tsx (hunk 12 above)
Comment (shreeyash07) [reply app/components/CustomLegend/index.tsx]: done

File: app/components/CustomLegend/index.tsx (hunk 13 above)
Comment (shreeyash07) [reply app/components/CustomLegend/index.tsx]: Done

File: app/views/Dashboard/CombinedIndicators/styles.css (hunk 14 above)
Comment (shreeyash07) [reply app/views/Dashboard/CombinedIndicators/styles.css]: Done

**Review by AdityaKhatri — APPROVED** [review]

**Pull request merged**